map/util/lanelet2_map_preprocessor/src/merge_close_lines.cpp

```diff
@@ -105,66 +94,65 @@ bool areLinesSame(
     return false;
   }
 
-  double sum_distance = 0;
-  for (const auto & pt : line1) {
-    sum_distance += boost::geometry::distance(pt.basicPoint(), line2);
-  }
-  for (const auto & pt : line2) {
-    sum_distance += boost::geometry::distance(pt.basicPoint(), line1);
-  }
+  double sum_distance =
+    std::accumulate(line1.begin(), line1.end(), 0.0, [&line2](double sum, const auto & pt) {
+      return sum + boost::geometry::distance(pt.basicPoint(), line2);
+    });
+  sum_distance +=
+    std::accumulate(line2.begin(), line2.end(), 0.0, [&line1](double sum, const auto & pt) {
+      return sum + boost::geometry::distance(pt.basicPoint(), line1);
+    });
 
-  double avg_distance = sum_distance / (line1.size() + line2.size());
+  double avg_distance = sum_distance / static_cast<double>(line1.size() + line2.size());
   std::cout << line1 << " " << line2 << " " << avg_distance << std::endl;
-  if (avg_distance < 1.0) {
-    return true;
-  } else {
-    return false;
-  }
+  return avg_distance < 1.0;
 }
 
-lanelet::BasicPoint3d getClosestPointOnLine(
+lanelet::BasicPoint3d get_closest_point_on_line(
   const lanelet::BasicPoint3d & search_point, const lanelet::ConstLineString3d & line)
 {
-  auto arc_coordinate = lanelet::geometry::toArcCoordinates(to2D(line), to2D(search_point));
+  auto arc_coordinate = lanelet::geometry::toArcCoordinates(
+    lanelet::utils::to2D(line), lanelet::utils::to2D(search_point));
   std::cout << arc_coordinate.length << " " << arc_coordinate.distance << std::endl;
-  return get3DPointFrom2DArcLength(line, arc_coordinate.length).basicPoint();
+  return get3_d_point_from2_d_arc_length(line, arc_coordinate.length).basicPoint();
 }
 
-lanelet::LineString3d mergeTwoLines(
+lanelet::LineString3d merge_two_lines(
   const lanelet::LineString3d & line1, const lanelet::ConstLineString3d & line2)
 {
   lanelet::Points3d new_points;
   for (const auto & p1 : line1) {
-    lanelet::BasicPoint3d p1_basic_point = p1.basicPoint();
-    lanelet::BasicPoint3d p2_basic_point = getClosestPointOnLine(p1, line2);
+    const lanelet::BasicPoint3d & p1_basic_point = p1.basicPoint();
+    lanelet::BasicPoint3d p2_basic_point = get_closest_point_on_line(p1, line2);
     lanelet::BasicPoint3d new_basic_point = (p1_basic_point + p2_basic_point) / 2;
     lanelet::Point3d new_point(lanelet::utils::getId(), new_basic_point);
     new_points.push_back(new_point);
   }
-  return lanelet::LineString3d(lanelet::utils::getId(), new_points);
+  return lanelet::LineString3d{lanelet::utils::getId(), new_points};
 }
 
-void copyData(lanelet::LineString3d & dst, lanelet::LineString3d & src)
+void copy_data(lanelet::LineString3d & dst, const lanelet::LineString3d & src)
 {
-  lanelet::Points3d points;
   dst.clear();
-  for (lanelet::Point3d & pt : src) {
-    dst.push_back(pt);
+  for (const lanelet::ConstPoint3d & pt : src) {
+    lanelet::Point3d pt_casted;
+    pt_casted = static_cast<lanelet::Point3d>(pt);
+    dst.push_back(pt_casted);
```

Comment:
Is there a specific reason you defined a temporary variable?
```suggestion
    dst.push_back(static_cast<lanelet::Point3d>(pt));
```

Reply:
Actually, no, but the reason why I used temporary variable here is to get rid of cppcheck's "don't use raw loop" notation.
If we use std::transform, it would be like
```
//for (const lanelet::ConstPoint3d & pt : src) {
//    dst.push_back(static_cast<lanelet::Point3d>(pt));
//}

std::transform(src.begin(), src.end(), std::back_inserter(dst), [](const lanelet::ConstPoint3d & pt : src) {
  return static_cast<lanelet::Point3d>(pt);
});
```
, but I think this is less readable and it raises some error which I was not able to solve.

Reply:
I understand the situation. I also think std::transform is less readable.
If you add a comment and explain the reason, I can approve this implementation :ok_man:

Reply:
actually, I found there is a suppression file which suppress the `useStlAlgorithm` (don't use raw loop) in autoware.universe repo: [file](https://github.com/autowarefoundation/autoware.universe/blob/main/.cppcheck_suppressions).
 so I fix it as you mentioned
 ```C++
 dst.push_back(static_cast<lanelet::Point3d>(pt));
 ```
 at 99843aa